python/ngen_cal/src/ngen/cal/agent.py

```diff
@@ -70,17 +74,14 @@ def __init__(self, model_conf, workdir: 'Path', log: bool=False, restart: bool=F
             # 0 correctly since not all basin params can be loaded.
             # There are probably some similar issues with explicit and independent, since they have
             # similar data semantics
-            workdirs = list(Path.glob(workdir, model_conf['type']+"_*_worker"))
+            workdirs = list(Path.glob(workdir, model_inner.type+"_*_worker"))
             if len(workdirs) > 1:
                 print("More than one existing workdir, cannot restart")
             elif len(workdirs) == 1:
-                self._job = JobMeta(model_conf['type'], workdir, workdirs[0], log=log)
+                self._job = JobMeta(model_inner.type, workdir, workdirs[0], log=log)
 
         if self._job is None:
-            self._job = JobMeta(model_conf['type'], workdir, log=log)
```

Comment:
These steps are minutely important to the use of the Agent, and I don't think they can be dropped...The agent manipulates the configs to ensure root/workdirs are set appropriately under the automatically created agent working dirs.

Reply:
I don't think they are dropped? I was just pulling them from the `pydantic` model instance now instead of the dictionary.

Reply:
Sorry, this comment ended up being unclear, it was intended for lines 81-83 where workdir and binary are configured relative to the Agent's dynamically created workdir.  These are critical for correct operation of agents especially in PSO where multiple agents are running simultaneously.

As noted in disscussion with @aaraney the current `resolved_binary` isn't actually working as intended, and is its own bug which can be resolved in this context.